lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2480,20 +2480,12 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
     return mapValue(BV,
                     transRelational(static_cast<SPIRVInstruction *>(BV), BB));
   case OpIAddCarry: {
-    IRBuilder Builder(BB);
     auto *BC = static_cast<SPIRVBinary *>(BV);
-    return mapValue(BV, Builder.CreateBinaryIntrinsic(
-                            Intrinsic::uadd_with_overflow,
-                            transValue(BC->getOperand(0), F, BB),
-                            transValue(BC->getOperand(1), F, BB)));
+    return mapValue(BV, transBuiltinFromInst("__spirv_IAddCarry", BC, BB));
```

Comment:
Remove `auto *BC = static_cast<SPIRVBinary *>(BV);` and
```suggestion
    return mapValue(BV, transSPIRVBuiltinFromInst(
                           static_cast<SPIRVInstruction *>(BV), BB));
```

Reply:
Doesn't `transSPIRVBuiltinFromInst` calls `transBuiltinFromInst` inside after going though switch clause and few ifs which, in that case, are never taken?